### ISSUE TITLE
No way to assert if a function call contained atomic code

### ISSUE THREAD

**Issue by antoinealb** [cvra/platform-abstraction#28]
It would be useful to have a way to test if a function did some atomic calls and verify if they were polite enough to release the locks. For example : 

``` c
TEST(...)
{
foo();
CHECK(atomic_count() > 0);
CHECK_FALSE(atomic_still_aquired());
}
```

**Comment by pierluca** [comment]
Too generic. Please clarify. What do you define as "atomic calls" ?
Mutex ? critical section ? another sync primitive? All of them?

**Comment by antoinealb** [comment]
Sorry, I meant a critical section.

For other sync primitives, I can already check for their usage using the `acquired_count` field.

**Comment by pierluca** [comment]
Given they're macros rather than proper functions working on a struct, I don't really know how to implement this in a meaningful way. Global variable in mock header doesn't seem appropriate because it would be shared between all critical sections being tested.

Will think about it but I'd be happy to hear your ideas.

**Comment by antoinealb** [comment]
A global variable could work, if we reset its state between each test, no ?

**Comment by antoinealb** [comment]
Another option I see would be to do it using a mock object and recording expectations. Not a huge fan of this because it makes the tests more fragile (you test the implementation not the behaviour).

**Comment by antoinealb** [comment]
I have a working implementation based on Cppumock. I will push it tomorrow after I reviewed it carefully.

**Comment by antoinealb** [comment]
fixed in #31 .
